jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/writer/AbstractOutputStreamWriter.java

```diff
@@ -0,0 +1,128 @@
+//
+// ========================================================================
+// Copyright (c) 1995 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.io.writer;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.Writer;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+
+import org.eclipse.jetty.util.ByteArrayOutputStream2;
+
+/**
+ * An alternate to {@link java.io.OutputStreamWriter} that supports
+ * several optimized implementation for well known {@link Charset}s,
+ * specifically {@link StandardCharsets#UTF_8} and {@link StandardCharsets#ISO_8859_1}.
+ */
+public abstract class AbstractOutputStreamWriter extends Writer
+{
+    /**
+     * Obtain a new {@link Writer} that converts characters written to bytes
+     * written to an {@link OutputStream}.
+     * @param outputStream The {@link OutputStream} to write to/
+     * @param charset The {@link Charset} name.
+     * @return A Writer that will
+     * @throws IOException If there is a problem creating the {@link Writer}.
+     */
+    public static Writer newWriter(OutputStream outputStream, String charset)
+        throws IOException
+    {
+        if (StandardCharsets.ISO_8859_1.name().equalsIgnoreCase(charset))
+            return new Iso88591Writer(outputStream);
+        if (StandardCharsets.UTF_8.name().equalsIgnoreCase(charset))
+            return new Iso88591Writer(outputStream);
+        return new EncodingWriter(outputStream, charset);
+    }
+
+    /**
+     * Obtain a new {@link Writer} that converts characters written to bytes
+     * written to an {@link OutputStream}.
+     * @param outputStream The {@link OutputStream} to write to/
+     * @param charset The {@link Charset}.
+     * @return A Writer that will
+     * @throws IOException If there is a problem creating the {@link Writer}.
+     */
+    public static Writer newWriter(OutputStream outputStream, Charset charset)
+        throws IOException
+    {
+        if (StandardCharsets.ISO_8859_1 == charset)
+            return new Iso88591Writer(outputStream);
+        if (StandardCharsets.UTF_8.equals(charset))
+            return new Iso88591Writer(outputStream);
+        return new EncodingWriter(outputStream, charset);
+    }
+
+    protected final int _maxWriteSize;
+    protected final OutputStream _out;
+    protected final ByteArrayOutputStream2 _bytes;
+    protected final char[] _chars;
```

Comment:
Make them `private`.

Reply:
But they are protected. They are used by the implementations, at least some are.